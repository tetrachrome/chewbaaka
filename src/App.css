body {
  -webkit-font-smoothing: antialiased;
  -moz-font-smoothing: grayscale;
}

@media only screen and (min-width: 768px) {
  .ui.container {
    width: 700px;
    max-width: 700px;
  }

  nav.ui.container .ui.header {
    float: left;
    margin-top: 12px;
  }

  nav.ui.container .ui.menu {
    float: right;
  }
}

.ui.inverted.vertical.center.aligned.segment {
  padding-top: 36px;
}

.ui.menu.topNavBarMenu {
  background-color: rgba(229, 239, 249, 0.65);
}

nav.ui.container .ui.menu .item {
  padding-bottom: 0;
}

nav.ui.container .ui.menu .item {
  font-size: 16px;
  padding-left: 0px;
  padding-right: 0px;
  padding-bottom: 14px;
  margin-left: 16px;
  margin-right: 16px;
  border-radius: 0;
}

/* nav.ui.container .ui.menu .active.item {
  background-color: inherit;
  border-bottom-color: rgb(51, 51, 51);
  border-bottom-style: solid;
  border-bottom-width: 2px;
} */

.ui.inverted.vertical.segment {
  height: 100vh;
  background-size: cover !important;
  background-position: center; /* Center the image */
}

.ui.inverted.vertical.segment.centerSegmentStyle1 {
  background-image: url('CCF_Field_and_Research_Centre.jpg');
}

.ui.inverted.vertical.segment.centerSegmentStyle2 {
  background-image: url('cheetah-mother-and-cubs-on-termite-mound.jpg');
}

.ui.inverted.vertical.segment.centerSegmentStyle3 {
  background-image: url('background_CCF_03.jpg');
}

.ui.inverted.vertical.segment.centerSegmentStyle4 {
  background-image: url('background_CCF_04.jpg');
}

.ui.content.container {
  height: 300px;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  margin: auto;
}

.ui.content.container h1.header {
  font-size: 36px;
  color: rgb(01, 01, 01);
}

.ui.content.container p {
  font-size: 21px;
  line-height: 1.6;
  font-weight: 300;
  font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
}

.ui.inverted.vertical.segment footer.ui.segment {
  height: 80px;
  width: 100%;
  position: absolute;
  bottom: 0;
  color: gray;
}

.ui.inverted.vertical.segment footer.ui.segment a {
  color: white;
}