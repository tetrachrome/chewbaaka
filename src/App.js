/**
 * App.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jun 30, 2020
 * Updated  : Jul 19, 2020
 */

import React from 'react'

import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom'

import HomePage from './components/homepage/HomePage'

import HistoryPage from './components/history/HistoryPage'
import BiologyPage from './components/biology/BiologyPage'
import EcologyPage from './components/ecology/EcologyPage'
import FuturePage from './components/future/FuturePage'

import TestPage from './components/shared/TestPage'

export default class App extends React.Component {
  render() {
    return (
      <Router>
        <Switch>
          { /** Route for testing components in development mode. */ }
          <Route exact path="/test">
            <TestPage />
          </Route>
          <Route exact path='/history'>
            <HistoryPage />
          </Route>
          <Route exact path='/biology'>
            <BiologyPage />
          </Route>
          <Route exact path='/ecology'>
            <EcologyPage />
          </Route>
          <Route exact path='/future'>
            <FuturePage />
          </Route>
          <Route exact path="/" >
            <HomePage />
          </Route>
          <Redirect to="/" />
        </Switch>
      </Router>
    )
  }
}
