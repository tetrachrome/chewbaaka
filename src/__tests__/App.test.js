/**
 * App.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jun 30, 2020
 * Updated  : Aug 04, 2020
 */

import React from 'react'

import { render, screen } from '@testing-library/react'

import renderer from 'react-test-renderer'

import App from '../App'

test('renders app', () => {
  render(
    <App />
  );

  // Tests home page component is present.
  const homePageComponent = screen.getByTestId("HomePageTestId");
  expect(homePageComponent).toBeInTheDocument();
});

test('App component snapshot', () => {
  const tree = renderer
    .create(
      <App />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
