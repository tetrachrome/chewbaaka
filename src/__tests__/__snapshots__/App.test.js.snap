// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot 1`] = `
<div
  className="HomePage"
  data-testid="HomePageTestId"
>
  <div
    className="ui top fixed TopNavBarMenu menu"
    data-testid="TopNavBarComponentTestId"
  >
    <div
      className="ui container"
    >
      <a
        className="header item"
        data-testid="TopNavBarComponentMainNavBarItemTestId"
        href="https://cheetah.org"
        onClick={[Function]}
      >
        <img
          className="ui small image"
          src="cheetah-conservation-fund-30-logo.svg"
          style={
            Object {
              "marginRight": "1.5em",
            }
          }
        />
      </a>
      <a
        className="item"
        href="https://www.iucnredlist.org/species/219/50649567"
        onClick={[Function]}
        target="_blank"
      >
        IUCN Status
      </a>
      <a
        className="item"
        href="https://wildnet.org/wildlife-programs/cheetah-namibia/"
        onClick={[Function]}
        target="_blank"
      >
        Conservation
      </a>
      <a
        className="item"
        href="https://cheetah.org/get-involved/ways-to-give/"
        onClick={[Function]}
        target="_blank"
      >
        Get Involved
      </a>
    </div>
  </div>
  <div
    className="ui vertical center aligned segment LandingPageGatewayOuterContainer LandingPageGatewayBackgroundStyle1"
    data-testid="LandingPageGatewayComponentTestId"
  >
    <div
      className="ui container LandingPageGatewayCoreContentContainer"
    >
      <h1
        className="ui inverted header"
      >
        Learn about cheetahs
      </h1>
      <h2
        className="ui inverted header"
        data-testid="LandingPageGatewayComponentSubtitlePartTestId"
      >
        Explore the Cheetah Museum at Cheetah Conservation Fund
      </h2>
      <a
        href="#Explore"
      >
        <button
          className="ui huge basic inverted button"
          onClick={[Function]}
        >
          Explore
        </button>
      </a>
    </div>
  </div>
  <div
    data-testid="ExploreGatewayComponentTestId"
    id="Explore"
  >
    <div
      className="ExploreGatewayIntro"
      data-testid="ExploreGatewayIntroTestId"
    >
      <div
        className="VerticalCushionPadding"
      >
        <img
          alt="CCF Cheetah Museum"
          className="ExploreGatewayIntroCCFCheetahMuseumImage"
          src="CCF_Cheetah_Museum.jpg"
        />
      </div>
      <div
        className="VerticalCushionPadding"
      >
        <div
          className="ui container"
          width={8}
        >
          <div
            className="ui huge center aligned header"
          >
            Explore CCF’s Cheetah Museum in Namibia and learn about the cheetahs, all at the comfort of your home.
          </div>
        </div>
      </div>
      <div
        className="ExploreGatewayIntroCenterGrid VerticalCushionPadding"
      >
        <div
          className="FluidTwoColumnContainerOuterContainer"
        >
          <div
            className="FluidTwoColumnContainerColumnContainer"
          >
            <div
              className="ExploreGatewayIntroCenterGridLhsColumnStyle"
            >
              <p
                className="CCFMissionStatementQuote"
              >
                “CCF is Changing the World to Save the Cheetah” - Dr. Laurie Marker
              </p>
            </div>
          </div>
          <div
            className="FluidTwoColumnContainerColumnContainer"
          >
            <div
              className="ExploreGatewayIntroCenterGridRhsColumnStyle"
            >
              <div
                className="ui inverted segment"
              >
                <p
                  className="JoinTheRace"
                >
                  Join the race to 
                  <a
                    href="https://twitter.com/search?q=%23SaveTheCheetah"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <span
                      className="SafeTheCheetah"
                    >
                      #SafeTheCheetah
                    </span>
                  </a>
                </p>
                <p>
                  Cheetahs can't win without us.
                </p>
                <a
                  className="ui basic inverted button"
                  href="https://cheetah.org/get-involved/ways-to-give/"
                  onClick={[Function]}
                  role="button"
                >
                  Get Involved
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ui container stackable center aligned four column grid"
      data-testid="ExploreGatewayGridTestId"
    >
      <div
        className="column ExploreGatewayGridItemColumn"
      >
        <div
          className="ExploreGatewayGridItemImageWrapper"
        >
          <a
            className="ExploreGatewayGridItemImageAnchor"
            data-testid="ExploreGatewayGridItemImageAnchorTestId"
            href="history"
          >
            <img
              className="ui medium image ExploreGatewayGridItemImage"
              data-testid="ExploreGatewayGridItemImageTestId"
              src="explore_gateway_grid_item_img_01.jpg"
            />
          </a>
        </div>
        <h2
          className="ui left aligned header ExploreGatewayGridItemHeader"
        >
          History
        </h2>
      </div>
      <div
        className="column ExploreGatewayGridItemColumn"
      >
        <div
          className="ExploreGatewayGridItemImageWrapper"
        >
          <a
            className="ExploreGatewayGridItemImageAnchor"
            data-testid="ExploreGatewayGridItemImageAnchorTestId"
            href="biology"
          >
            <img
              className="ui medium image ExploreGatewayGridItemImage"
              data-testid="ExploreGatewayGridItemImageTestId"
              src="explore_gateway_grid_item_img_02.jpg"
            />
          </a>
        </div>
        <h2
          className="ui left aligned header ExploreGatewayGridItemHeader"
        >
          Biology
        </h2>
      </div>
      <div
        className="column ExploreGatewayGridItemColumn"
      >
        <div
          className="ExploreGatewayGridItemImageWrapper"
        >
          <a
            className="ExploreGatewayGridItemImageAnchor"
            data-testid="ExploreGatewayGridItemImageAnchorTestId"
            href="ecology"
          >
            <img
              className="ui medium image ExploreGatewayGridItemImage"
              data-testid="ExploreGatewayGridItemImageTestId"
              src="explore_gateway_grid_item_img_03.jpg"
            />
          </a>
        </div>
        <h2
          className="ui left aligned header ExploreGatewayGridItemHeader"
        >
          Ecology
        </h2>
      </div>
      <div
        className="column ExploreGatewayGridItemColumn"
      >
        <div
          className="ExploreGatewayGridItemImageWrapper"
        >
          <a
            className="ExploreGatewayGridItemImageAnchor"
            data-testid="ExploreGatewayGridItemImageAnchorTestId"
            href="future"
          >
            <img
              className="ui medium image ExploreGatewayGridItemImage"
              data-testid="ExploreGatewayGridItemImageTestId"
              src="explore_gateway_grid_item_img_04.jpg"
            />
          </a>
        </div>
        <h2
          className="ui left aligned header ExploreGatewayGridItemHeader"
        >
          Future
        </h2>
      </div>
    </div>
    <br />
  </div>
  <div
    className="ui inverted segment"
    data-testid="FooterComponentTestId"
    style={
      Object {
        "padding": "5em 0em",
      }
    }
  >
    <div
      className="ui container"
    >
      <div
        className="ui inverted stackable divided grid"
      >
        <div
          className="row"
        >
          <div
            className="three wide column"
          >
            <img
              className="ui small image"
              src="cheetah-conservation-fund-logo.jpg"
            />
          </div>
          <div
            className="six wide column"
          >
            <h4
              className="ui inverted header"
            >
              Resources
            </h4>
            <div
              className="ui inverted link list"
              role="list"
            >
              <a
                className="item"
                href="https://cheetah.org/about/what-we-do/"
                onClick={[Function]}
                role="listitem"
                target="_blank"
              >
                About CCF
              </a>
              <a
                className="item"
                href="https://animalfactguide.com/animal-facts/cheetah/"
                onClick={[Function]}
                role="listitem"
                target="_blank"
              >
                Cheetah Fun Facts
              </a>
              <a
                className="item"
                href="https://wildnet.org/wildlife-programs/cheetah-namibia/"
                onClick={[Function]}
                role="listitem"
                target="_blank"
              >
                Wildlife Conservation Network & CCF
              </a>
              <a
                className="item"
                href="https://www.iucnredlist.org/species/219/50649567"
                onClick={[Function]}
                role="listitem"
                target="_blank"
              >
                IUCN Red List - Cheetah
              </a>
            </div>
          </div>
          <div
            className="seven wide column"
          >
            <h4
              className="ui inverted header"
            >
              Cheetah Conservation Fund
            </h4>
            <p
              data-testid="footerSectionCCF"
            >
              <a
                href="https://cheetah.org/"
              >
                Cheetah Conservation Fund
              </a>
               is a 501(c)(3) nonprofit organization, dedicated to saving the cheetahs in the wild and changing the world to be a better place.
            </p>
            <p>
              All textual content presented on this site are copyright of Cheetah Conservation Fund.
            </p>
            <p>
              This site is not an affiliation of Cheetah Conservation Fund.
            </p>
            <p>
              Made with 
              <span
                aria-label="heart"
                role="img"
              >
                ❤️
              </span>
               by 
              Will Li
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
